src/Razor/src/Microsoft.VisualStudio.LanguageServer.ContainedLanguage/DefaultLSPDocumentManager.cs

```diff
@@ -101,6 +101,11 @@ public override void UntrackDocument(ITextBuffer buffer)
             {
                 Debug.Fail($"Couldn't remove {uri.AbsolutePath}. This should never ever happen.");
             }
+
+            foreach (VirtualDocument virtualDocument in lspDocument.VirtualDocuments)
+            {
+                virtualDocument.Dispose();
```

Comment:
Hmm, should we have the top-level lsp document do this logic? Aka instead of iterating through the virtual documents here you'd do `lspDocument.Dispose()` which would iterate through its own virtual documents and dispose.

Reply:
Sounds good, I think it makes sense. 

---
In reply to: [451971134](https://github.com/dotnet/aspnetcore-tooling/pull/2201#discussion_r451971134) [](ancestors = 451971134)